src/mscorlib/shared/System/Memory.cs

```diff
@@ -281,4 +295,23 @@ private static int CombineHashCodes(int h1, int h2, int h3)
         }
 
     }
-}
\ No newline at end of file
+
+    internal sealed class MemoryDebugView<T>
```

Comment:
You do not need this type. You can use ReadOnlyMemoryDebugView for both Memory and ReadOnlyMemory.

(Notice that it is what you are sort of doing anyway already because of the _memory field below is ReadOnlyMemory actually.)